### PR TITLE
Avoid crashing where a sane alternative exists

### PR DESCRIPTION
**Pull request by rsheeter**
Trying to define ops with degenerate cases in ways that aren't raise an error.

### CODE REVIEW

File: src/picosvg/svg_types.py [hunk 1]
```diff
@@ -70,24 +70,33 @@ def _copy_common_fields(
...
         x1, y1, x2, y2 = svg_pathops.bounding_box(self.as_cmd_seq())
         return Rect(x1, y1, x2 - x1, y2 - y1)
 
     def apply_transform(self, transform: Affine2D) -> "SVGPath":
-        cmds = svg_pathops.transform(self.as_cmd_seq(), transform)
         target = self.as_path()
         if target is self:
             target = copy.deepcopy(target)
+        cmds = (('M', (0, 0)),)
+        if not transform.is_degenerate():
```
Comment (anthrotype) [inline src/picosvg/svg_types.py:98]: why not do a simple if-else here? Most of the time we're throwing that `cmds` object away, as the transform is rarely degenerate.
E.g.
```python
if transform.is_degenerate():
   cmds = (('M', (0, 0)),)
else:
   cmds = svg_pathops.transform(self.as_cmd_seq(), transform)
```

File: src/picosvg/svg_types.py [hunk 2]
```diff
@@ -70,24 +70,33 @@ def _copy_common_fields(
...
+        """False if we're sure this shape will not paint. True if it *might* paint."""
         def _visible(fill, opacity):
             return fill != "none" and opacity != 0
             # we're ignoring fill-opacity
 
-        return _visible(self.fill, self.opacity) or _visible(
-            self.stroke, self.stroke_opacity
-        )
+        if _visible(self.stroke, self.stroke_opacity):
+            return True
```
Comment (anthrotype) [inline src/picosvg/svg_types.py:80]: a stroke that only has a move also doesn't paint, no?

File: src/picosvg/svg_types.py (hunk 1 above)
Comment (rsheeter) [reply src/picosvg/svg_types.py:98]: x = default
if something
  x = something else

is shorter and I like, in a language lacking a signal like `if likely(...)`to have the main branch of the if be the likely path

File: src/picosvg/svg_types.py (hunk 2 above)
Comment (rsheeter) [reply src/picosvg/svg_types.py:80]: what kind of person would make a stroke with only a move :D

Comment (rsheeter) [reply src/picosvg/svg_types.py:80]: Shuffled; now we return false if all cmds are move

**Pull request merged**